src/main/java/com_github_leetcode/ListNode.java

```diff
@@ -16,6 +16,12 @@ public ListNode(int val, ListNode next) {
         this.next = next;
     }
 
+    @Override
+    public boolean equals(Object obj) {
+        return this.val == ((ListNode) obj).val;
```

Comment:
We compare ListNode by pointers not by value. The custom comparator may be applied for a solution.